src/main/java/group1/mips_simulator/components/ExecutionInstructions.java

```diff
@@ -0,0 +1,191 @@
+package group1.mips_simulator.components;
+
+import group1.mips_simulator.components.cpuParts.ConditionCode;
+import group1.mips_simulator.components.cpuParts.Register;
+import group1.mips_simulator.components.instructionParts.Field;
+import group1.mips_simulator.components.instructionParts.RXIA_Instruction;
+
+/**
+ * A class that holds all the functions for actually executing an Instruction object
+ */
+public class ExecutionInstructions {
+
+    /**
+     * SETCCE r
+     * opcode 44(octal)
+     * If c(r) = 0, the E bit of the condition code is set to 1,
+     * else the E bit of the condition code is set to 0
+     */
+    public void execute_setcce(Computer computer, RXIA_Instruction i) {
+        Register targetR = computer.cpu.regfile.getGPR(i.getR().value);
+        Value regContents = targetR.read();
+        boolean newEBit = (regContents.get() == 0); // if c(r) == 0, then newEBit is true (1).
+        computer.cpu.regfile.getCC().setBit(ConditionCode.EBIT_INDEX, newEBit);
+    }
+
+    /**
+     * JZ x, address[,I]
+     * 06(octal)
+     * Jump If Zero:
+     * If the E bit of  condition code is 1, then PC <− EA
+     * Else PC <- PC+1
+     */
+    public void execute_jz(Computer computer, RXIA_Instruction i) {
+        boolean eBit = computer.cpu.regfile.getCC().getEBit();
+        if (eBit) {
+            // if eBit is 1
+            // PC <-- EA
+            Value ea = computer.calculateEA(i);
```

Comment:
My general philosophy is that "Value" objects are the representation of this thing in here. I agree that a short type would be easier to manage (because primitives are nicer than objects). I don't really care which we return I guess.